ktor-http/src/io/ktor/http/URLBuilder.kt

```diff
@@ -38,9 +36,9 @@ class URLBuilder(
         }
         out.append(host)
 
-        if (port != protocol.defaultPort) {
+        if (portOrDefault != protocol.defaultPort) {
             out.append(":")
-            out.append(port.toString())
+            out.append(portOrDefault.toString())
```

Comment:
We could try switch to the different logic here:
   - if the port is default: don't print anything.
   - if the user specifies port by hand: print it(even it equals the default value)